perfkitbenchmarker/publisher.py

```diff
@@ -144,12 +143,24 @@ def AddMetadata(self, metadata, benchmark_spec):
 
       if vm.scratch_disks:
         data_disk = vm.scratch_disks[0]
+        # Legacy metadata keys
         metadata[name_prefix + 'scratch_disk_type'] = data_disk.disk_type
         metadata[name_prefix + 'scratch_disk_size'] = data_disk.disk_size
         metadata[name_prefix + 'num_striped_disks'] = (
             data_disk.num_striped_disks)
-        if data_disk.disk_type == disk.PIOPS:
+        if getattr(data_disk, 'iops', None) is not None:
           metadata[name_prefix + 'scratch_disk_iops'] = data_disk.iops
+        # Modern metadata keys
+        metadata[name_prefix + 'data_disk_0_type'] = data_disk.disk_type
+        metadata[name_prefix + 'data_disk_0_size'] = (
+            data_disk.disk_size * data_disk.num_striped_disks)
+        metadata[name_prefix + 'data_disk_0_num_stripes'] = (
+            data_disk.num_striped_disks)
+        if getattr(data_disk, 'iops', None) is not None:
```

Comment:
I duplicated the statements so I could keep the legacy and modern metadata handling completely separate. I think the duplication is worth it to make it very clear what is legacy, and also to make it very easy to remove the legacy code once we're ready to do that. If you still disagree, I'll merge the if statements, but that's why I did this.